src/wp-includes/class-wp-scripts.php

```diff
@@ -435,11 +438,12 @@ public function do_item( $handle, $group = false ) {
 
 		$tag  = $translations . $cond_before . $before_script;
 		$tag .= sprintf(
-			"<script%s src='%s' id='%s-js'%s></script>\n",
+			"<script%s src='%s' id='%s-js'%s%s></script>\n",
 			$this->type_attr,
 			esc_url( $src ),
 			esc_attr( $handle ),
-			$strategy ? " {$strategy}" : ''
+			$strategy ? " {$strategy}" : '',
+			$strategy_data_attr ? " data-wp-strategy='{$strategy_data_attr}'" : ''
```

Comment:
Perhaps we should only add this if the strategy has changed?

```suggestion
			$strategy_has_changed ? " data-wp-intended-strategy='{$intended_strategy}'" : ''
```

Or would it be helpful to include it if the fallback behavior is not happening?

Reply:
@westonruter the way I had it prior (in my conditional checks further down) only did it if the strategy had changed, but I think it makes sense to have it regardless as it'll allow us to more granularly determine who is using the new API, and also probably offer more field research opportunities?

Reply:
I think it is more consistent to always include the intended strategy when it is not the default (blocking), regardless of whether it has changed. Also, just to be clear, the description of this PR says (emphasis mine):

> Tracks the **final** script strategy in a data-wp-strategy data attribute when using delayed loading strategies, useful for being able to query http archive data for future field analysis.

However, if I'm following correctly, we are actually including the _intended_ (i.e., registered) strategy. The final (i.e. eligible) strategy will be added to the script itself using either the `async` or `default` attribute. The main reason for having this value is so developers can more easily see that the final strategy has been modified after the script was registered.

Reply:
Agreed @joemcgill I think I've made this over complicated by trying to track the strategy changes, ignoring blocking etc. I'm going to resolve conflicts and output the intended strategy (blocking included if none was assigned) regardless, it makes the most sense.